src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -888,6 +887,33 @@ public static function get_resolved_theme_uris( $theme_json ) {
 			$resolved_theme_uris[] = $resolved_theme_uri;
 		}
 
+		// Block styles.
+		if ( ! empty( $theme_json_data['styles']['blocks'] ) ) {
+			foreach ( $theme_json_data['styles']['blocks'] as $block_name => $block_styles ) {
+				if ( ! isset( $block_styles['background']['backgroundImage']['url'] ) ) {
```

Comment:
Ah yes, probably. Then we could skip the line below: `$background_image_url = $block_styles['background']['backgroundImage']['url'] ?? null;` 

👍🏻 Good one!

Reply:
Actually, I take that back. The only advantage here is a short circuit out of the loop, and there's a string check below, so it can probably stay.

But it does mean the `isset( $background_image_url ) &&` below is superfluous.

So I'd replace it with something like this:

```php
				if ( ! isset( $block_styles['background']['backgroundImage']['url'] ) ) {
					continue;
				}
				$background_image_url = $block_styles['background']['backgroundImage']['url'];
				if (
					is_string( $background_image_url ) &&
					// Skip if the src doesn't start with the placeholder, as there's nothing to replace.
					str_starts_with( $background_image_url, $placeholder )
				) {
```